lib/src/main/java/com/dynatrace/metric/util/Metric.java

```diff
@@ -203,13 +206,23 @@ public Builder setDimensions(DimensionList dimensions) {
 
     /**
      * (Optional) Set the timestamp for the exported metric line. In most cases, {@link
-     * Builder#setCurrentTime()} should be suitable.
+     * Builder#setCurrentTime()} should be suitable. If the timestamp is from before the year 2000
+     * or from after the year 3000, the timestamp will be discarded and no value will be set.
      *
      * @param timestamp an {@link Instant} object describing the time at which the {@link Metric}
      *     was created.
      * @return this
      */
     public Builder setTimestamp(Instant timestamp) {
+      int year = timestamp.atZone(ZoneOffset.UTC).getYear();
+      if (year < 2000 || year > 3000) {
+        logger.warning(
+            "Timestamp does not seem to be in millisecond format. Skipping setting timestamp, the current server time will be added upon ingestion.");
```

Comment:
Users can create their Instant from any time unit as long as they use the correct `Instant.ofEpoch*` method. It's opaque to them whether millis are used under to hood or something else.